src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -2275,6 +2275,30 @@ public void updateTypes(Map<String, String> typeToCorrect) {
     }
   }
 
+  /**
+   * Based on the set returned by JavaTypeCorrect, this method corrects the Throwable extension for
+   * synthetic classes as needed.
+   *
+   * @param typesToExtendThrowable the set of synthetic types that need Throwable extension.
+   */
+  public void updateTypesToExtendThrowable(Set<String> typesToExtendThrowable) {
+    for (String typeToExtendThrowable : typesToExtendThrowable) {
+      Iterator<UnsolvedClassOrInterface> iterator = missingClass.iterator();
+      while (iterator.hasNext()) {
+        UnsolvedClassOrInterface missedClass = iterator.next();
+        // Class comparison is based on class name and package name only.
```

Comment:
The comparison below is only based on the simple name - not on the package. I think we could pretty easily construct a test where this matters - for example, does the following behave correctly (if it's a target method)?:
```
org.packageone.MyThrowable get() throws org.packagetwo.MyThrowable { return null; }
```

(It might be necessary to move around the order of the actual throwable and the fake throwable, because this loop returns after it finds the first one - so if it finds the right one first, this code will continue to work.)

In general, I don't think we should do class comparisons based on simple names only if at all possible, and this is an example where we should use the fully-qualified name, regardless of whether you can get a test like the above to fail (although I strongly suggest that you try to do so, and add a test for that case regardless).

Reply:
Professor,

It's quite difficult to use whole qualified names because the error messages in Javac only tell us the simple name of types.

I don't think this current approach can cause any compilation error. The worst case scenario is that if we have two synthetic classes with the same name, and one of them should extends `Throwable`, Specimin will make both of them extend `Throwable`. Specimin doesn't support extension for synthetic classes, apart from `extends Exception` and `extends Throwable`. So we should not have any conflict.

Reply:
> The worst case scenario is that if we have two synthetic classes with the same name, and one of them should extends Throwable, Specimin will make both of them extend Throwable

I see your point here. Please add a test case to demonstrate this imprecision, at least, so that we have documentation of Specimin's behavior in this scenario.